src/vs/workbench/contrib/notebook/browser/viewModel/notebookOutlineEntryFactory.ts

```diff
@@ -41,12 +43,25 @@ function getMarkdownHeadersInCellFallbackToHtmlTags(fullContent: string) {
 	return headers;
 }
 
-export class NotebookOutlineEntryFactory {
+export const INotebookOutlineEntryFactory = createDecorator<INotebookOutlineEntryFactory>('INotebookOutlineEntryFactory');
 
-	private cellOutlineEntryCache: Record<string, entryDesc[]> = {};
+export interface INotebookOutlineEntryFactory {
+	readonly _serviceBrand: undefined;
+
+	getOutlineEntries(cell: ICellViewModel, index: number): OutlineEntry[];
+	cacheSymbols(cell: ICellViewModel, cancelToken: CancellationToken): Promise<void>;
+}
+
+export class NotebookOutlineEntryFactory implements INotebookOutlineEntryFactory {
+
+	declare readonly _serviceBrand: undefined;
+
+	private cellOutlineEntryCache: Record<string, { version: number; entries: entryDesc[] }> = {};
```

Comment:
Do we still need the version here? Seems like the cache out of date concept was added in an early commit and removed later on.